resources/user-agents/apps/facebook/facebook-app.json

```diff
@@ -24,80 +24,67 @@
       },
       "children": [
         {
-          "match": "Mozilla/5.0 (#DEVICE##PLATFORM#) applewebkit* (*khtml*like*gecko*)*Mobile*FB*",
+          "match": "Mozilla/5.0 (#DEVICE##PLATFORM#) applewebkit* (*khtml*like*gecko*)*Mobile*FBForIPhone*",
           "devices": {
-            "iPhone": "iPhone",
             "iPad": "iPad",
+            "iPhone": "iPhone",
             "iPod": "iPod Touch"
           },
           "platforms": [
-            "iOS_A_10_3", "iOS_A_10_2", "iOS_A_10_1", "iOS_A_10_0",
-            "iOS_A_9_3", "iOS_A_9_2", "iOS_A_9_1", "iOS_A_9_0",
-            "iOS_A_8_4", "iOS_A_8_3", "iOS_A_8_2", "iOS_A_8_1", "iOS_A_8_0",
-            "iOS_A_7_1", "iOS_A_7_0", "iOS_A_6_1", "iOS_A_6_0", "iOS_A_5_1", "iOS_A_5_0",
-            "iOS_A_4_3", "iOS_A_4_2", "iOS_A_4_1", "iOS_A_4_0", "iOS_A_3_2", "iOS_A_3_1", "iOS_A_3_0",
+            "iOS_A_7_1", "iOS_A_7_0",
+            "iOS_A_6_1", "iOS_A_6_0",
+            "iOS_A_5_1", "iOS_A_5_0",
+            "iOS_A_4_3", "iOS_A_4_2", "iOS_A_4_1", "iOS_A_4_0",
+            "iOS_A_3_2", "iOS_A_3_1", "iOS_A_3_0",
+            "iOS_A"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (#PLATFORM#) applewebkit* (*khtml*like*gecko*)*Mobile*FBForIPhone*",
+          "platforms": [
             "iOS_A",
-            "iOS_B",
-            "iOS_C_10_3", "iOS_C_10_2", "iOS_C_10_1", "iOS_C_10_0",
-            "iOS_C_9_3", "iOS_C_9_2", "iOS_C_9_1", "iOS_C_9_0",
-            "iOS_C_8_4", "iOS_C_8_3", "iOS_C_8_2", "iOS_C_8_1", "iOS_C_8_0",
-            "iOS_C_7_1", "iOS_C_7_0", "iOS_C_6_1", "iOS_C_6_0", "iOS_C_5_1", "iOS_C_5_0", "iOS_C_4_3",
             "iOS_C"
           ]
         },
         {
-          "match": "Mozilla/5.0 (#DEVICE##PLATFORM#) AppleWebKit (*khtml*like*gecko*) Mobile ?FB*",
```

Comment:
As I mentioned in the issue description, this "FB" syntax would pick up agents like these:

>Mozilla/5.0 (iPhone; CPU iPhone OS 9_3_4 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Mobile/13G35 baiduboxapp/0_01.5.2.8_enohpi_1002_5211/4.3.9_1C2%257enohPi/1099a/A9FB3E33A9AEAB890EEAB08E548D043A6524E7D61OCITRFCDQF/1
>Mozilla/5.0 (iPhone; CPU iPhone OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Mobile/14FBB1

Which are not Facebook :)